blog/content/post/2020-07-21-secure-app-coding-practices.md

```diff
@@ -0,0 +1,74 @@
+
+### Always validate the data also on the server
+
+The "required" attribute for the text field or "disabled" for the select item, we are able to turn off these items in the development tools. Validations written in JS can also be modified on the fly, or you can simply turn off JavaScript.
+After setting the "E-mail" and "Password" fields as required in HTML, validation is very easy to bypass. It is enough to enter the inspector, change a few elements and the form can be sent even if the validation conditions are not met. Therefore, treat the frontend validation as an improvement to UX, which is to be a help for the user. In terms of security, server-side validation is much more important. The user is not able to turn it off and at best can look for vulnerabilities to somehow cheat it.
+Remember - always verify the data also the second time, on the server side.
+
+### Try… catch mistakes
+
+Working with external libraries, these may have different error notification systems. Some just "throw" Error, others print exact messages, and some let us decide ourselves how to behave in case of a problem. However, we do not want JS throwing the default error content to the user, because it is often incomprehensible and can expose us to leakage of sensitive data. Remember to use the Try… catch block to contol the errors.
+
+### Keep your web app updated
+
+Try to use popular and valued packages. Pay attention to the number of editions, downloads and the date of the last update. It is very important that the package is still supported. Then, even if any gaps are found, they should be patched quickly.
+
+Avoid packages that the author himself describes as deprecated. Even if you think a package is perfect for your needs, it's quite possible that in many cases it won't work properly with your code. In addition, there is a good chance that it has some security flaws. Remember that IT is an extremely dynamic industry.
+Update your dependent packages
```

Comment:
Should it be a heading?

Reply:
fixed

Reply:
patronising words: try, remember that and so on...